django/contrib/gis/db/models/query.py

```diff
@@ -675,9 +676,10 @@ def _geocol_select(self, geo_field, field_name):
         if geo_field not in opts.fields:
             # Is this operation going to be on a related geographic field?
             # If so, it'll have to be added to the select related information
-            # (e.g., if 'location__point' was given as the field name).
+            # (e.g., if 'location__point' was given as the field name, then
+            # chop the non-relational field and add select_related('location')).
             # Note: the operation really is defined as "must add select related!"
-            self.query.add_select_related([field_name])
+            self.query.add_select_related([field_name.rsplit(LOOKUP_SEP)[0]])
```

Comment:
Correct, I meant it to be `rsplit(LOOKUP_SEP, 1)`. I'll see about adding a test for that case.